libflux/src/semantic/types.rs

```diff
@@ -139,7 +139,22 @@ impl PolyType {
             .join(" + ")
     }
     fn equal(&self, poly: &PolyType) -> bool {
-        self.vars == poly.vars && self.cons == poly.cons && self.expr == poly.expr
+        self.vars == poly.vars && self.expr == poly.expr && self.cons.len() == poly.cons.len() && {
+            for (tvar, kinds) in self.cons.iter() {
+                if let Some(pkinds) = poly.cons.get(tvar) {
+                    let mut kinds = kinds.clone();
+                    let mut pkinds = pkinds.clone();
+                    kinds.sort();
+                    pkinds.sort();
+                    if kinds != pkinds {
+                        return false;
+                    }
+                } else {
+                    return false;
+                }
+            }
+            true
+        }
```

Comment:
bug fix for equality

Reply:
Thanks for this. I'm wondering if we should remove the display logic that sorts kinds since that made this bug kind of confusing?